src/util-napatech.c

```diff
@@ -0,0 +1,755 @@
+
+NapatechCurrentStats NapatechGetCurrentStats(uint16_t id) 
+{
+
+    return current_stats[id];
+}
+
+
+static uint16_t TestStreamConfig(
+        NtInfoStream_t hInfo,
+        NtStatStream_t hStatStream,
+        NapatechStreamConfig stream_config[], 
+        uint16_t num_inst) 
+{
+    int status;
+    char buffer[80]; // Error buffer
+    NtStatistics_t stat; // Stat handle.
+
+    uint16_t num_active = 0;
+    
+    for (uint16_t inst = 0; inst < num_inst; ++inst) {
+
+        // Check to see if it is an active stream
+        memset(&stat, 0, sizeof (NtStatistics_t));
+
+        // Read usage data for the chosen stream ID
+        stat.cmd = NT_STATISTICS_READ_CMD_USAGE_DATA_V0;
+        stat.u.usageData_v0.streamid = (uint8_t) stream_config[inst].stream_id;
+
+        if ((status = NT_StatRead(hStatStream, &stat)) != NT_SUCCESS) {
+            // Get the status code as text
+            NT_ExplainError(status, buffer, sizeof (buffer));
+            SCLogError(SC_ERR_NAPATECH_INIT_FAILED, "NT_StatRead():2 failed: %s\n", buffer);
+            return 0;
+        }
+
+        if (stat.u.usageData_v0.data.numHostBufferUsed > 0) {
+            stream_config[inst].is_active = true;
+            num_active++;
+        } else {
+            stream_config[inst].is_active = false;
+        }
+    }
+
+    return num_active;
+}
+
+
+
+static uint32_t updateStreamStats(ThreadVars *tv,
+        NtInfoStream_t hInfo,
+        NtStatStream_t hStatStream,
+        uint16_t num_streams,
+        NapatechStreamConfig stream_config[],
+        StreamCounters streamCounters[]
+        ) 
+{
+    static uint64_t rxPktsStart[MAX_STREAMS] = {0};
+    static uint64_t rxByteStart[MAX_STREAMS] = {0};
+    static uint64_t dropStart[MAX_STREAMS] = {0};
+
+    int status;
+    char error_buffer[80]; // Error buffer
+    NtInfo_t hStreamInfo;
+    NtStatistics_t hStat; // Stat handle.
+
+    // Query the system to get the number of streams currently instantiated
+    hStreamInfo.cmd = NT_INFO_CMD_READ_STREAM;
+    if ((status = NT_InfoRead(hInfo, &hStreamInfo)) != NT_SUCCESS) {
+        NT_ExplainError(status, error_buffer, sizeof (error_buffer) - 1);
+        SCLogError(SC_ERR_NAPATECH_INIT_FAILED, "NT_InfoRead() failed: %s\n", error_buffer);
+        exit(EXIT_FAILURE);
+    }
+    
+    uint16_t num_active;
+    if ((num_active = TestStreamConfig(hInfo, hStatStream, stream_config, num_streams)) == 0) {
+        // None of the configured streams are active
+        return 0;
+    } 
+    
+    // At least one stream is active so proceed with the stats.    
+    uint16_t inst_id = 0;
+    uint32_t stream_cnt = 0;
+    for (stream_cnt = 0; stream_cnt < num_streams; ++stream_cnt) {
+
+
+        while(inst_id < num_streams) {
+            if (stream_config[inst_id].is_active) {
+                break;
+            } else {
+                ++inst_id;
+            }
+        }        
+        if (inst_id == num_streams) break;
```

Comment:
please use 2 lines as stepping through this code with gdb will be harder otherwise